src/main/java/org/aszjch/demoapp/domain/article/ArticleServiceImpl.java

```diff
@@ -0,0 +1,73 @@
+package org.aszjch.demoapp.domain.article;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.aszjch.demoapp.domain.article.port.ArticlePublisher;
+import org.aszjch.demoapp.domain.article.port.ArticleRepository;
+import org.aszjch.demoapp.domain.articlefile.ArticleFile;
+import org.aszjch.demoapp.domain.articlefile.port.ArticleFileAssignmentService;
+
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+@RequiredArgsConstructor
+public class ArticleServiceImpl implements ArticleService, ArticleFileAssignmentService {
+
+    private final ArticleRepository repository;
+    private final ArticlePublisher publisher;
+
+    @Override
+    public List<Article> get() {
+        return repository.findAll();
+    }
+
+    @Override
+    public Article create(final Article article) {
+        final Article saved = repository.save(article);
+        publisher.publish(new ArticleCreatedDto(saved.getId(), saved.getTitle(), saved.getAuthor(),
+                                                saved.getAbstractText()));
+        return saved;
+    }
+
+    @Override
+    public Optional<Article> getById(final Long id) {
+        return repository.findById(id);
+    }
+
+    @Override
+    public Article update(final Long id, final Article article) {
+        final Article entity = getById(id).orElse(new Article());
+        entity.setId(id);
+        entity.setTitle(article.getTitle());
+        entity.setAbstractText(article.getAbstractText());
+        entity.setCreationDate(article.getCreationDate());
+        entity.setAuthor(article.getAuthor());
+        entity.setAbstractText(article.getAbstractText());
+        return repository.save(entity);
+    }
+
+    @Override
+    public void delete(final Long id) {
+        repository.deleteById(id);
+    }
+
+    @Override
+    public void assignFile(final ArticleFile articleFile) {
```

Comment:
Nie wiem czy nie chcemy jakiejś zwrotki z tego. To się powinno zachowywać tak samo jak update tylko trochę taki specyficzny. Zastanowimy się nad tym wspólnie jak już będą testy i reszta. Dodaj gdzieś do listy tematów do dyskusji.